src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/actions/DeleteAction.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexstatemanagement.actions
+
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.ManagedIndexMetaData
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.actions.DeleteActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.steps.AttemptDeleteStep
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.steps.Step
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+
+class DeleteAction(
+    clusterService: ClusterService,
+    client: Client,
+    managedIndexMetaData: ManagedIndexMetaData,
+    config: DeleteActionConfig
+) : Action(type, managedIndexMetaData) {
+
+    private val attemptDeleteStep =
+            AttemptDeleteStep(clusterService, client, config, managedIndexMetaData)
+    private val steps = listOf(attemptDeleteStep)
+
+    override fun getSteps(): List<Step> = steps
+
+    override fun getNextStep(managedIndexMetaData: ManagedIndexMetaData): Step {
```

Comment:
shall we return null if the current step is AttempDeleteStep?

Reply:
The name "getNextStep" might be confusing. There should never be a step after AttemptDeleteStep because for that to succeed we would of had to delete the index successfully which should delete this job.

This method is returning the step that should be executed during the execution of runJob.

Will change to getStepToExecute